Web Quiz Engine (Kotlin)/task/src/engine/service/QuizService.kt

```diff
@@ -6,34 +6,62 @@ import engine.model.QuizPostResponseBodyCorrect
 import engine.model.QuizPostResponseBodyIncorrect
 import engine.exception.QuizNotFoundException
 import engine.entity.Quiz
+import engine.model.QuizDto
 import org.springframework.stereotype.Service
 
 @Service
-class QuizService(private val quizRepository: WebQuizRepository) {
+class QuizService(
+    private val quizRepository: WebQuizRepository,
+) {
 
-    fun findQuizById(
-        id: Int,
-        omitId: Boolean = false,
-        omitAnswerList: Boolean = false
-        ) =
-        quizRepository.getQuizById(
-            id = id,
-            omitId = omitId,
-            omitAnswerList = omitAnswerList,
-    )?: throw QuizNotFoundException(quizId = id)
+    fun findQuizById(id: Int): QuizDto =
+        quizRepository
+            .findById(id)
+            .orElseThrow { QuizNotFoundException(quizId = id) }
+            .let {
+                with(it) {
+                    QuizDto(
+                        id = id,
+                        title = title,
+                        text = text,
+                        optionList = optionList,
+                    )
+                }
+            }
 
-    fun findQuizList(omitAnswerList: Boolean = false) =
-        quizRepository.getAllQuizList(omitAnswerList = omitAnswerList)
+    fun findQuizList(): List<QuizDto> =
+        quizRepository.findAll().map { quiz ->
+            with(quiz) {
+                QuizDto(
+                    id = id,
+                    title = title,
+                    text = text,
+                    optionList = optionList,
+                )
+            }
+        }
 
     fun checkQuizSolution(quizId: Int, answerList: List<Int>?): QuizPostResponseBody =
-        quizRepository.getQuizById(id = quizId)?.let { quiz ->
+        quizRepository.findById(quizId)
+            .orElseThrow { QuizNotFoundException(quizId = quizId)}
+            .let { quiz ->
             if ((quiz.answerList.isNullOrEmpty() && answerList.isNullOrEmpty()) ||
                 (quiz.answerList?.containsAll(answerList.orEmpty()) == true &&
```

Comment:
`containsAll` is an expensive operation, especially if your list gets bigger.
Instead of running it twice, I would just first do a size comparison, and then do a `containsAll`.

Reply:
Thks, I need to get better at thinking about complexity.
Done as 
```kotlin
if ((quiz.answerList.isNullOrEmpty() && answerList.isNullOrEmpty()) ||
                quiz.answerList?.size == answerList?.size &&
                (quiz.answerList?.containsAll(answerList.orEmpty()) == true &&
                        answerList?.containsAll(quiz.answerList) == true))
```